### ISSUE TITLE
[Bug] Zettlr does not update metadata of all files after changing ID regular expression

### ISSUE THREAD

**Issue by ailintom** [Zettlr/Zettlr#2810]
### Description

When you change ID regular expression in the Preferences to switch to another ID format, Zettlr continues to use old IDs, until you manually force updating metadata for each file. (This can be done for example by editing and saving the file).

Instead, Zettlr should immediately update metadata for all files to switch to the new ID formats after the user changes the ID regular expression in the Preferences.

### Reproducing

1) Create a new workspace using the default ID regular expression.

2) Create a new file named foo.md  and type in an ID in the default format "20211113112630" ).

3) Create another file named boo.md a type in a custom ID "ZK00001N"

4) Save both files.

5) Change ID regular expression in Preferences to the pattern matching the second file: "(ZK\d{5}N)"

5) Create a new file and start typing the link start ("[["). A pop-up list shows up listing the available files.

6) Expected behaviour: foo.md is shown in the pop-up list only with username; boo.md with the ID "ZK00001N" and with username.

Actual behaviour:  In the pop-up list, foo.md is shown with the now obsolete ID ( 20211113112630); boo.md is shown in the list without the ID. 

7) If you manually link boo.md by using its new ID [[ZK00001N]], the link does not work ("The requested file was not found" is shown instead of the file content). 

8) Only if you edit the boo.md by adding a few random characters to it and saving the file, it's ID is updated in the stored metadata, and the link  [[ZK00001N]] in the third file becomes working. 

9) The same with foo.md, although its id is now invalid, it still does work as a link (the file content is shown in the pop-up if you type in the link [[20211113112630]]) until you manually edit foo.md and save it.

### Zettlr Version

Stable (most recent version)

### Specify version

2.0.3

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Windows 10

**Comment by ailintom** [comment]
Supposedly, Zettlr should empty the cache and reopen all open dirs upon changing the "ID regular expression" setting in the preferences.

**Comment by ailintom** [comment]
It was discussed earlier in #1746 that this is a non-issue because you should never change your ID's. However, suppose, you have two sets of .md files from different sources using two different ID formats and you want to use keep ID formats (so as to not change the IDs) by using an ID regular expression of the type ( format1 | format2 ). Hence, I believe it still is a bug, and the program should empty the cache automatically upon changing the ID regular expression,

**Comment by nathanlesage** [comment]
I mean, I can provide a menu item to do this cache clearing easily and we can in the docs add an explanation that you need to clear the cache?

**Comment by ailintom** [comment]
That surely would make the process much more transparent for those who are unaware of what happens under the hood. 
However, why not also automatically perform a cache clearing after changing certain settings that affect the extraction of metadata (ID regular expression, Link start, Link ending, Count characters instead of words [if I am right that word count is also part of the metadata])?

**Comment by nathanlesage** [comment]
> However, why not also automatically perform a cache clearing after changing certain settings that affect the extraction of metadata

Because that's not the main problem. The main problem is that the complete FSAL tree has to be re-loaded in that case. The cache just persists this stuff across sessions. Restarting the app will always work, but I think a simple refresh method on the FSAL might do the job.

**Comment by nathanlesage** [comment]
> I would like to point out that from my experience with Beta 3 "Check Markdown for style issues" is another option requiring a restart and it is currently not on the list of options requiring restart.

That should not require a restart. Does it begin to work when you type in your document? It should then begin to work.

**Comment by ailintom** [comment]
My problem is that it does not stop working after switching off, unless I restart Zettlr.

**Comment by ailintom** [comment]
@nathanlesage I rechecked and I can confirm: I performed a clean install of Zettlr 3.0 Beta 3. I disabled  "Check Markdown for style issues" in preferences. But the style checks are still there 
![image](https://user-images.githubusercontent.com/24455000/231674919-c00c57fe-f3e7-458d-b9aa-6249f9ae71b4.png)
until I restart Zettlr. 
![image](https://user-images.githubusercontent.com/24455000/231675116-8c21db43-1d62-43dc-b60e-0edfee57be64.png)
